### PR TITLE
ENH: Remove local dataset definitions from histopathology models

### PR DESCRIPTION
**Pull request by ant0nsc**

### CODE REVIEW

**Review by dccastro — APPROVED** [review]

File: hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py [hunk 1]
```diff
@@ -40,8 +39,6 @@ def __init__(self, **kwargs: Any) -> None:
             is_finetune=False,
             is_caching=True,
             num_top_slides=0,
-            # declared in DatasetParams:
-            local_datasets=[Path("/tmp/datasets/TCGA-CRCk")],
             azure_datasets=["TCGA-CRCk"],
```
Comment (kenza-bouzid) [inline hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py:42]: can we remove hardcoded strings? 
`from histopathology.datasets.default_paths import TCGA_CRCK_DATASET_ID`

File: hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILEPanda.py [hunk 2]
```diff
@@ -135,8 +130,6 @@ def __init__(self, **kwargs: Any) -> None:
             tile_size=224,
             random_offset=True,
             background_val=255,
-            # declared in DatasetParams:
-            local_datasets=[Path("/tmp/datasets/PANDA")],
```
Comment (kenza-bouzid) [inline hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILEPanda.py:139]: same here with `PANDA_DATASET_ID`?

**Review by kenza-bouzid — APPROVED** [review]

**Review by kenza-bouzid — APPROVED** [review]

File: hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py (hunk 1 above)
Comment (ant0nsc) [reply hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py:42]: The default path is still used in some tests, will leave that as is.

File: hi-ml-histopathology/src/histopathology/scripts/tcga_dataset_prep.py [hunk 3]
```diff
@@ -19,7 +20,7 @@ def check_dataset_csv_paths(dataset_dir: Path) -> None:
 if __name__ == '__main__':
     # Script needs to be started in the parent folder of the dataset folder
     current_dir = Path.cwd()
-    expected_datasetdir = "TCGA-Crck"
+    expected_datasetdir = TCGA_CRCK_DATASET_ID
```
Comment (kenza-bouzid) [inline hi-ml-histopathology/src/histopathology/scripts/tcga_dataset_prep.py:23]: Heads up: `TCGA_CRCK_DATASET_ID = "TCGA-CRCk"` id different from old `expected_datasetdir = "TCGA-Crck"` 
case mismatch...

**Pull request merged**